scipy/stats/_hypotests.py

```diff
@@ -956,6 +959,210 @@ def barnard_exact(table, alternative="two-sided", pooled=True, n=32):
     return BarnardExactResult(wald_stat_obs, p_value)
 
 
+BoschlooExactResult = make_dataclass(
+    "BoschlooExactResult", [("statistic", float), ("pvalue", float)]
+)
+
+
+def boschloo_exact(table, alternative="two-sided", n=32):
+    r"""Perform Boschloo's exact test on a 2x2 contingency table.
+
+    Parameters
+    ----------
+    table : array_like of ints
+        A 2x2 contingency table.  Elements should be non-negative integers.
+
+    alternative : {'two-sided', 'less', 'greater'}, optional
+        Defines the null and alternative hypotheses. Default is 'two-sided'.
+        Please see explanations in the Notes section below.
+
+    n : int, optional
+        Number of sampling points used in the construction of the sampling
+        method. Note that this argument will automatically be converted to
+        the next higher power of 2 since `scipy.stats.qmc.Sobol` is used to
+        select sample points. Default is 32. Must be positive. In most cases,
+        32 points is enough to reach good precision. More points comes at
+        performance cost.
+
+    Returns
+    -------
+    ber : BoschlooExactResult
+        A result object with the following attributes.
+
+        statistic : float
+            The Fisher statistic.
```

Comment:
Is the statistic of Boschloo's test, which it the p-value of Fisher's test, commonly referred to as "The Fisher statistic"?

This looks like a mistake. To be unambiguous, maybe "Boschloo's test statistic; that is, the p-value from Fisher's exact test".

Reply:
Would you like me to propose a maintenance PR with the changes ?

Reply:
You could do this now, we can wait for another round of CI.

Reply:
Done in 721384b79babe716748d97e00ee5762acbcb1f25
Also updated the test pattern in 1a4fe9a4e38598f6b0e322a783bc5e0224ccb3a1